interpreter/interpreter.go

```diff
@@ -658,6 +676,13 @@ func (f function) Call(argsObj values.Object) (values.Value, error) {
 	return v, nil
 }
 func (f function) doCall(args Arguments) (values.Value, error) {
+	if f.itrp == nil {
```

Comment:
Why was this check needed? Does a test case need to be added?

Reply:
It is possible for a function to be called with no interpreter. The old implementation didn't use the interpreter for type information, but the new one does. Basically, all of the calls to `itrp` inside of function could be null for a decently long time and we just got lucky.